### PR TITLE
Anki21 port

### PR DESCRIPTION
**Pull request by mikehardy**
Hi there - I use (or used to use) this add-on with Anki 2.0, and I thought a port to 2.1 would be a good thing

This seems to work for me, although if you're not careful in testing Google will stop responding to you because it thinks you are a robot - https://github.com/sloria/TextBlob/issues/215

### CODE REVIEW

**Comment by mikehardy** [comment]
@prz3m - just in case you don't have github notifications turned on by default, I think this will send an email :-) ? Cheers

File: __init__.py [hunk 1]
```diff
@@ -183,17 +185,21 @@ def getDBPath():
...
         raise IOError
-    db_path = unicode(db_path)
+    db_path = str(db_path)
     return db_path
 
 
 def getKindleVocabPath():
     try:
-        if platform == "win32":
+        if platform.system == "Windows":
```
Comment (prz3m) [inline __init__.py]: `platform` is actually a string, so this should be changed back to `if platform == "win32":` (https://docs.python.org/3.6/library/sys.html#sys.platform)

File: __init__.py [hunk 2]
```diff
@@ -183,17 +185,21 @@ def getDBPath():
...
             for l in string.ascii_uppercase:
                 path = r"{}:\\system\vocabulary\vocab.db".format(l)
                 if os.path.exists(path):
                     return r"{}:\\system\vocabulary".format(l)
+        elif platform.system == "Darwin":
+            path = "/Volumes/Kindle/system/vocabulary/vocab.db"
+            if os.path.exists(path):
+                return path
```
Comment (prz3m) [inline __init__.py]: This function should return directory path, not db file path

**Review by prz3m — CHANGES_REQUESTED** [review]
Thank you for your work!

File: __init__.py (hunk 1 above)
Comment (mikehardy) [reply __init__.py]: ah - I see - stackoverflow led me astray :-), sorry. I'll work on this bit

File: __init__.py (hunk 2 above)
Comment (mikehardy) [reply __init__.py]: interesting - I'll check this out as well. I was just attempting to follow the pattern of the other ones but didn't have my kindle mounted at the time so it wasn't working (making for a bad test). I'll fix it up and make sure this bit actually works

**Comment by prz3m** [comment]
Hi Mike! I must admit that I've stopped using Anki some time ago and I didn't even notice that new version was released. Thank you for your contribution, there's only one small issue with detecting db file on connected Kindle.
I'm very grateful for this PR, have a great new year!

**Comment by prz3m** [comment]
I commited changes to `getKindleVocabPath`, there was an old mistake in windows path (double backslash after drive letter, I don't know why I wrote it that way) which apparently wasn't causing trouble in previous Anki version. Sorry for messing around on your branch :smile:

**Review by prz3m — APPROVED** [review]

**Comment by mikehardy** [comment]
Awesome thanks!

**Pull request merged**